src/transformStateWithClones.js

```diff
@@ -5,9 +5,47 @@
  * @param {Object[]} actions
  *
  * @return {Object[]}
+ *
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  /**  write code here
+  const stateHistory = [];
+  const copyState = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(copyState, action.extraData);
+    } else if (action.type === 'removeProperties') {
+      for (const element of action.keysToRemove) {
+        delete copyState[element];
+      }
+    } else if (action.type === 'clear') {
+      for (const key of Object.keys(copyState)) {
+        delete copyState[key];
+      }
+    }
+  */
+
+  const stateHistory = [];
+  const stateCopy = { ...state };
```

Comment:
The stateCopy is created outside of the loop, which means it will be mutated with each action. Instead, you should create a new copy of the state within the loop before applying changes to ensure that the original state object is not mutated.

Reply:
ok